x/pki/keeper/msg_server_update_pki_revocation_distribution_point.go

```diff
@@ -0,0 +1,161 @@
+		return pkitypes.NewErrCRLSignerCertificatePidNotEqualMsgPid("pid in updated CRL Signer Certificate must be equal to pid in revocation point")
+	}
+
+	if pid == 0 && pid != revocationPointPid {
+		return pkitypes.NewErrPidNotFound("pid not found in updated CRL Signer Certificate when it is provided in revocation point")
+	}
+
+	return nil
+}
+
+func verifyUpdatedCertificate(updatedCertificate string, revocationPoint types.PkiRevocationDistributionPoint, isPrevCertPAA bool) error {
+	updatedCrlSignerCertificate, err := x509.DecodeX509Certificate(updatedCertificate)
+	if err != nil {
+		return pkitypes.NewErrInvalidCertificate(err)
+	}
+
+	if isPrevCertPAA {
+		err = verifyUpdatedPAA(updatedCrlSignerCertificate, revocationPoint.Vid)
+	} else {
+		err = verifyUpdatedPAI(updatedCrlSignerCertificate, revocationPoint.Vid, revocationPoint.Pid)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k msgServer) UpdatePkiRevocationDistributionPoint(goCtx context.Context, msg *types.MsgUpdatePkiRevocationDistributionPoint) (*types.MsgUpdatePkiRevocationDistributionPointResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+
+	signerAccount, _ := k.dclauthKeeper.GetAccountO(ctx, signerAddr)
+
+	// check if signer has vendor role
+	if !k.dclauthKeeper.HasRole(ctx, signerAddr, dclauthtypes.Vendor) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized,
+			"MsgUpdatePkiRevocationDistributionPoint transaction should be signed by an account with the \"%s\" role",
+			dclauthtypes.Vendor,
+		)
+	}
+
+	pkiRevocationDistributionPoint, isFound := k.GetPkiRevocationDistributionPoint(ctx, msg.Vid, msg.Label, msg.IssuerSubjectKeyID)
+	if !isFound {
+		return nil, pkitypes.NewErrPkiRevocationDistributionPointDoesNotExists("PKI revocation distribution point does not exist")
+	}
+
+	crlSignerCertificate, err := x509.DecodeX509Certificate(pkiRevocationDistributionPoint.CrlSignerCertificate)
+	if err != nil {
+		return nil, pkitypes.NewErrInvalidCertificate(err)
+	}
+
+	if pkiRevocationDistributionPoint.Vid != signerAccount.VendorID {
+		return nil, sdkerrors.Wrap(pkitypes.ErrCRLSignerCertificateVidNotEqualAccountVid,
+			"MsgUpdatePkiRevocationDistributionPoint signer must have the same vid as provided in an existing certificate from the revocation point",
+		)
+	}
+
+	if msg.CrlSignerCertificate != "" {
+		err = verifyUpdatedCertificate(msg.CrlSignerCertificate, pkiRevocationDistributionPoint, crlSignerCertificate.IsSelfSigned())
+
+		if err != nil {
+			return nil, err
+		}
+
+		pkiRevocationDistributionPoint.CrlSignerCertificate = msg.CrlSignerCertificate
+	}
+
+	CRLRevocationType := uint32(1)
+	if pkiRevocationDistributionPoint.RevocationType == CRLRevocationType && (msg.DataFileSize != 0 || msg.DataDigest != "" || msg.DataDigestType != 0) {
```

Comment:
It can be check on the static validation step (in the message). The corresponding test should be in static validation tests, no need for it in `handler_tests`.